core/src/test/scala/integration/kafka/api/PlaintextAdminIntegrationTest.scala

```diff
@@ -1948,6 +1948,166 @@ class PlaintextAdminIntegrationTest extends BaseAdminIntegrationTest {
+      )).all().get()
+      waitForTopics(client, List(testTopicName), List())
+
+      val producer = createProducer()
+      try {
+        producer.send(new ProducerRecord(testTopicName, 0, null, null)).get()
+      } finally {
+        Utils.closeQuietly(producer, "producer")
+      }
+
+      val testGroupId = "test_group_id"
+      val testClientId = "test_client_id"
+      val fakeGroupId = "fake_group_id"
+
+      def createProperties(): Properties = {
+        val newConsumerConfig = new Properties(consumerConfig)
+        newConsumerConfig.setProperty(ConsumerConfig.GROUP_ID_CONFIG, testGroupId)
+        newConsumerConfig.setProperty(ConsumerConfig.CLIENT_ID_CONFIG, testClientId)
+        newConsumerConfig
+      }
+
+      val consumerSet = Set(createShareConsumer(configOverrides = createProperties()))
+      val topicSet = Set(testTopicName)
+
+      val latch = new CountDownLatch(consumerSet.size)
+      try {
+        def createShareConsumerThread[K,V](consumer: ShareConsumer[K,V], topic: String): Thread = {
+          new Thread {
+            override def run : Unit = {
+              consumer.subscribe(Collections.singleton(topic))
+              try {
+                while (true) {
+                  consumer.poll(JDuration.ofSeconds(5))
+                  if (latch.getCount > 0L)
+                    latch.countDown()
+                  consumer.commitSync()
+                }
+              } catch {
+                case _: InterruptException => // Suppress the output to stderr
+              }
+            }
+          }
+        }
+
+        // Start consumers in a thread that will subscribe to a new group.
+        val consumerThreads = consumerSet.zip(topicSet).map(zipped => createShareConsumerThread(zipped._1, zipped._2))
+
+        try {
+          consumerThreads.foreach(_.start())
+          assertTrue(latch.await(30000, TimeUnit.MILLISECONDS))
+          // Test that we can list the new group.
+          TestUtils.waitUntilTrue(() => {
+            val matching = client.listShareGroups.all.get.asScala.filter(group =>
+              group.groupId == testGroupId &&
+                group.state.get == ShareGroupState.STABLE)
+            matching.size == 1
+          }, s"Expected to be able to list $testGroupId")
+
+          TestUtils.waitUntilTrue(() => {
+            val options = new ListShareGroupsOptions().inStates(Set(ShareGroupState.STABLE).asJava)
+            val matching = client.listShareGroups(options).all.get.asScala.filter(group =>
+              group.groupId == testGroupId &&
+                group.state.get == ShareGroupState.STABLE)
+            matching.size == 1
+          }, s"Expected to be able to list $testGroupId in state Stable")
+
+          TestUtils.waitUntilTrue(() => {
+            val options = new ListShareGroupsOptions().inStates(Set(ShareGroupState.EMPTY).asJava)
+            val matching = client.listShareGroups(options).all.get.asScala.filter(
```

Comment:
ditto this can be 
```
            client.listShareGroups(options).all.get.stream().filter(
              _.groupId == testGroupId).count() == 0
              ```